api/segments/serializers.py

```diff
@@ -156,6 +167,25 @@ def _update_or_create_segment_rules(
                 child_rules, rule=child_rule, is_create=is_create
             )
 
+    def _update_or_create_metadata(
```

Comment:
Do we have to do this because we are not using NestedModelSerializer Here?

Reply:
Yes.

Reply:
Do we know why are we not using NestedModelSerializer ?

Reply:
I think the decision to not employ NestedModelSerializer here is based on the simplicity and structure of the models and data relationships we are dealing with.